eZ/Publish/Core/Persistence/Legacy/Content/Gateway/DoctrineDatabase.php

```diff
@@ -120,7 +120,14 @@ public function getContext()
      */
     public function insertContentObject(CreateStruct $struct, $currentVersionNo = 1)
     {
-        $initialLanguageCode = $this->languageHandler->load($struct->initialLanguageId)->languageCode;
+        if (isset($struct->mainLanguageId)) {
+            $initialLanguageId = $struct->mainLanguageId;
```

Comment:
`$initialLanguageId = !empty($struct->mainLanguageId) ? $struct->mainLanguageId : $struct->initialLanguageId;`

and `$this->languageHandler->load($initialLanguageId)->languageCode;` can still be inline here unless you see other places that can use it